plugin/backend_unsupported.go

```diff
@@ -24,8 +25,11 @@ func (pm *Manager) Enable(name string, config *types.PluginEnableConfig) error {
 }
 
 // Inspect examines a plugin config
-func (pm *Manager) Inspect(name string) (tp types.Plugin, err error) {
-	return tp, errNotSupported
+func (pm *Manager) Inspect(refOrID string) (tp types.Plugin, err error) {
+	// Even though plugin is not supported, we still want to return `not found`
+	// error so that `docker inspect` (without `--type` specified) returns correct
+	// `not found` message
+	return tp, fmt.Errorf("no such plugin name or ID associated with %q", refOrID)
```

Comment:
@thaJeztah @jhowardmsft Yes the change should be made in `docker inspect` part, not the `plugin inspect` part. Sorry about that.

I created another PR #29212 to address `docker plugin inspect` issue. Please take a look.